sdk/storage/azure-storage-file-datalake/out/azure/storage/filedatalake/_data_lake_file_client.pyi

```diff
@@ -0,0 +1,28 @@
+# Stubs for azure.storage.filedatalake._data_lake_file_client (Python 3)
+#
+# NOTE: This dynamically typed stub was automatically generated by stubgen.
+
+from ._deserialize import process_storage_error
+from ._generated.models import StorageErrorException
+from ._models import FileProperties
+from ._path_client import PathClient
+from ._serialize import add_metadata_headers, get_access_conditions, get_mod_conditions, get_path_http_headers
+from ._shared.base_client import parse_connection_str
+from ._shared.request_handlers import get_length, read_length
+from ._shared.response_handlers import return_response_headers
+from ._shared.uploads import IterStreamer
+from ._upload_helper import upload_datalake_file
+from typing import Any, Optional
+
+class DataLakeFileClient(PathClient):
+    def __init__(self, account_url: str, file_system_name: str, file_path: str, credential: Optional[Any]=..., **kwargs: Any) -> None: ...
+    @classmethod
+    def from_connection_string(cls: Any, conn_str: str, file_system_name: str, file_path: str, credential: Optional[Any]=..., **kwargs: Any) -> DataLakeFileClient: ...
+    def create_file(self, content_settings: Optional[ContentSettings]=..., metadata: Optional[Dict[str, str]]=..., **kwargs: Any) -> Dict[str, Union[str, datetime]]: ...
+    def delete_file(self, **kwargs: Any) -> None: ...
+    def get_file_properties(self, **kwargs: Any) -> FileProperties: ...
+    def upload_data(self, data: Union[AnyStr, Iterable[AnyStr], IO[AnyStr]], length: Optional[int]=..., overwrite: Optional[bool]=..., **kwargs: Any) -> Dict[str, Any]: ...
+    def append_data(self, data: Union[AnyStr, Iterable[AnyStr], IO[AnyStr]], offset: int, length: Optional[int]=..., **kwargs: Any) -> Dict[str, Union[str, datetime, int]]: ...
+    def flush_data(self, offset: int, retain_uncommitted_data: Optional[bool]=..., **kwargs: Any) -> Dict[str, Union[str, datetime]]: ...
+    def read_file(self, offset: Optional[int]=..., length: Optional[int]=..., stream: Optional[IO]=..., **kwargs: Any) -> Union[int, byte]: ...
```

Comment:
If I recall in Blobs - we use the pattern:
```python
stream_response = client.download_blob()
with open(...) as io:
    stream_response.readinto(io)
```
for reading into an open stream - is there a reason we're not following that pattern here?
If I pass a stream in, are we still returning the byte data as well?

Reply:
If you pass in a stream, the downloaded data will be written to that stream and no data will be returned.
Ah I think because the API's name is read*, and it seems weird to return a downloader to customers

Reply:
We should not use different return types depending on a parameter, and should use the same behavior we have for downloading in the blobs library.
Maybe we could name this `download_file` or `download_data` (to match the upload function)?

Otherwise, if this must do the download as a single blocking call, then reading into a stream should be a separate method - not combined into this one.

Reply:
Rename to `dowload_file` and use the `download_blob` behaviour.